backend/danswer/connectors/google_drive/constants.py

```diff
@@ -1,7 +1,29 @@
-DB_CREDENTIALS_DICT_TOKEN_KEY = "google_drive_tokens"
-DB_CREDENTIALS_DICT_SERVICE_ACCOUNT_KEY = "google_drive_service_account_key"
-DB_CREDENTIALS_DICT_DELEGATED_USER_KEY = "google_drive_delegated_user"
+UNSUPPORTED_FILE_TYPE_CONTENT = ""  # keep empty for now
+DRIVE_FOLDER_TYPE = "application/vnd.google-apps.folder"
+DRIVE_SHORTCUT_TYPE = "application/vnd.google-apps.shortcut"
 
-BASE_SCOPES = ["https://www.googleapis.com/auth/drive.readonly"]
-FETCH_PERMISSIONS_SCOPES = ["https://www.googleapis.com/auth/drive.metadata.readonly"]
-FETCH_GROUPS_SCOPES = ["https://www.googleapis.com/auth/cloud-identity.groups.readonly"]
+FILE_FIELDS = "nextPageToken, files(mimeType, id, name, permissions, modifiedTime, webViewLink, shortcutDetails, owners)"
+SLIM_FILE_FIELDS = "nextPageToken, files(id, permissions(emailAddress, type), permissionIds, webViewLink)"
+FOLDER_FIELDS = "nextPageToken, files(id, name, permissions, modifiedTime, webViewLink, shortcutDetails)"
+USER_FIELDS = "nextPageToken, users(primaryEmail)"
+
+# these errors don't represent a failure in the connector, but simply files
+# that can't / shouldn't be indexed
+ERRORS_TO_CONTINUE_ON = [
+    "cannotExportFile",
+    "exportSizeLimitExceeded",
+    "cannotDownloadFile",
```

Comment:
What would you think about adding `notFound`?

Am trying to index a shared drive with a service account impersonating a workspace user, and indexing is hanging on these while they retry every 30s:
```
googleapiclient.errors.HttpError: <HttpError 404 when requesting https://www.googleapis.com/drive/v3/files/1YgCxrbFsFS1ZQBdsialmpe_jVs4MMBLmTJe9sV22uZw?supportsAllDrives=true&alt=json returned "File not found: 1YgCxrbFsFS1ZQBdsialmpe_jVs4MMBLmTJe9sV22uZw.". Details: "[{'message': 'File not found: 1YgCxrbFsFS1ZQBdsialmpe_jVs4MMBLmTJe9sV22uZw.', 'domain': 'global', 'reason': 'notFound', 'location': 'fileId', 'locationType': 'parameter'}]">
```

Reply:
Try again once this PR is in. The flow is:
If include_shared_drives is True, the service account impersonates all (or indicated) users and indexes their personal drive.
Then, to access shared drives, it impersonates the primary admin email to access all drives that admin has access to